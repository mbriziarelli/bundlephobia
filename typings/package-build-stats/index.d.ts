declare module "package-build-stats" {
  interface Asset {
    name: string
    type: string
    size: number
    gzip: number
    parse: unknown
  }

  interface DependencySize {
    name: string
    approximateSize: number
  }

  export interface PackageStats {
    dependencyCount: number
    hasJSNext: boolean
    hasJSModule: boolean
    hasSideEffects: boolean
    peerDependencies: string[]
    ignoredMissingDependencies?: string[]
    assets: Asset[]
    dependencySizes: DependencySize[]
    size: number
    gzip: number
    parse: unknown
  }

  export interface BuildPackageStatsOptions {
    client: "npm" | "yarn"
    limitConcurrency?: boolean
    networkConcurrency?: number
    customImports?: string[]
  }

  export type CustomErrorName = "PackageNotFoundError" | "InstallError" | "EntryPointError" | "MissingDependencyError" | "CLIBuildError" | "BuildError"

  export class CustomError {
    name: CustomErrorName
    originalError: Error
    extra?: unknown 
  }

  export default function (packageName: string, options: BuildPackageStatsOptions): Promise<PackageStats>
}