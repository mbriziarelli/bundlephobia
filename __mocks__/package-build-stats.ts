export default jest.genMockFromModule('package-build-stats')
