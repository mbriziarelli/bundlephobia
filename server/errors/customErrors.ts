import { CustomErrorName } from 'package-build-stats'
import { ServerError } from 'bundlephobia-errors'

export default {
  PackageNotFoundError: {
    statusCode: 400,
    code: 'PackageNotFoundError',
    message: 'Package not found.',
  },
  InstallError: {
    statusCode: 500,
    code: 'InstallError',
    message: 'Error while installing package.',
  },
  EntryPointError: {
    statusCode: 500,
    code: 'EntryPointError',
    message: 'Error while building package.',
  },
  MissingDependencyError: {
    statusCode: 500,
    code: 'MissingDependencyError',
    message: 'A package dependency is missing.',
  },
  CLIBuildError: {
    statusCode: 500,
    code: 'CLIBuildError',
    message: 'Building package from cli failed.',
  },
  BuildError: {
    statusCode: 500,
    code: 'BuildError',
    message: 'Building package failed.',
  },
} as Record<CustomErrorName, ServerError>
