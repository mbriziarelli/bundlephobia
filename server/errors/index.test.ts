import { CustomErrorName } from 'package-build-stats'
import customErrors from './customErrors'
import {
  makeBadPackageNameError,
  makeErrorFromCustomError,
  isCustomError,
} from '.'

describe('errors module', () => {
  describe('makeBadPackageNameError function', () => {
    it('returns a BadRequest error with message containing the bad package name', () => {
      const error = makeBadPackageNameError('error')
      expect(error).toMatchSnapshot()
    })
  })
  describe('makeErrorFromCustomError function', () => {
    it('returns the corresponding server error from the custom error', () => {
      Object.keys(customErrors).forEach(key => {
        expect(
          makeErrorFromCustomError({
            name: key as CustomErrorName,
            originalError: new Error(),
          })
        ).toBe(customErrors[key as CustomErrorName])
      })
    })
  })
  describe('isCustomError function', () => {
    it('returns true if the given parameter is a custom error', () => {
      Object.keys(customErrors).forEach(key => {
        expect(
          isCustomError({ name: customErrors[key as CustomErrorName].code })
        ).toBe(true)
      })
    })
    it('returns false if the given parameter is an object without the "name" property', () => {
      expect(isCustomError({})).toBe(false)
    })
    it('returns false if the given parameter is a primitive value', () => {
      ;[null, undefined, 0, 'string', false].forEach(value => {
        expect(isCustomError(value)).toBe(false)
      })
    })
  })
})
