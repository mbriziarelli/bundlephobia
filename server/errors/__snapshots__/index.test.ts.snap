// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors module makeBadPackageNameError function returns a BadRequest error with message containing the bad package name 1`] = `
Object {
  "code": "BadRequest",
  "message": "Invalid or empty package name: \\"error\\"",
  "statusCode": 400,
}
`;
